### DISCUSSION TITLE
`readonly` property on Slider

### DISCUSSION THREAD

**Discussion by dilawar** [slint-ui/slint#1442] [Ideas] [answered]
Does it make sense to add `readonly` property on the `Slider` widget https://slint-ui.com/releases/0.2.5/docs/rust/slint/docs/widgets/index.html#slider? If I am not wrong, another user-input widget  `Button` supports it.

In one of my use cases, I don't want the user to be able to move the slider until one of the background processes is complete. In another use case, I am abusing the `Slider` a bit and using it like a meter to _show_ a value between 0 and 100. In both cases, the ability to make the slider `readonly` will help.

**Comment by ogoffart** [comment, marked as answer]
I think it makes sense, yes.
Note that there is also a `enabled` property that can be set to false, but somehow it seems it isn't documented :-/

**Reply by dilawar** [reply]
`enabled` might just do the job! 👍🏼

**Reply by dilawar** [reply]
If I am not doing something wrong, the meaning of `enabled` is slightly different than what I was hoping. Once the `enabled` of the slider is set to `false`, any update to its value does not change the slider in the UI? The `readonly` will update the slider value and redraw it but won't accept the events generated by a user.

**Reply by dilawar** [reply]
Apparently, I was doing something wrong. Even when `enabled=false`, the slider gets redrawn when the value is updated.
